### DISCUSSION TITLE
Unreleased Rust version in Cargo.toml?

### DISCUSSION THREAD

**Discussion by linusg** [dani-garcia/vaultwarden#2117] [Q&A]
### Subject of the issue

When trying to compile vaultwarden from source, `cargo build` complains about the specified Rust version 1.57:

https://github.com/dani-garcia/vaultwarden/blob/016fe2269e3079be65f86c4887b959341c984bfa/Cargo.toml#L6

The latest Rust release appears to be 1.56.1: https://github.com/rust-lang/rust/blob/master/RELEASES.md - is this on purpose? The only scenario I can imagine where this works is compiling Rust itself from source and they already bumped the version number before the official release. Thanks!

### Deployment environment

N/A.

### Steps to reproduce

Install latest release of Rust. Try to build vaultwarden.

### Expected behaviour

Build is successful.

### Actual behaviour

Build is not successful.

**Comment by BlackDex** [comment]
The current version of Vaultwarden needs to be build with nightly. Also see the `rust-toolchain` file.

Nightly starts at 1.57, and currently is at 1.58.
